controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -234,6 +234,7 @@ func (r *BareMetalHostReconciler) Reconcile(ctx context.Context, request ctrl.Re
 		return ctrl.Result{}, errors.Wrap(err, "failed to check services availability")
 	}
 	if !ready {
+		provisionerNotReady.Inc()
```

Comment:
>In the provisioning backend we're currently essentially checking the availability of the Ironic Inspector and Conductor, so in case of issues (let's say the inspector will remain down for one minute), this counter will be increased for each host reconciliation loop, always increasing. 

Agree, we should avoid this situation.

>  maybe a gauge could be a better fit.

I am not very clear about your thoughts. Do you mean that only the message  `ironic is not available is not available` needs to be displayed?

Reply:
I'm referring to the [current type](https://github.com/metal3-io/baremetal-operator/pull/997/files#diff-2e6604b6d369e5e798174541ac76b46906672100105a15482d1cddfeced7bcb9R117) of `provisionerNotReady`, which is a `prometheus.Counter`, since it doesn't appear to be suitable for setting up an alert (the original problem).
Instead, I'm suggesting to evaluate the usage of another [Prometheus metric type ](https://prometheus.io/docs/concepts/metric_types/) to map more precisely the (Ironic) service status. It could be also useful to double check how the same problem have been eventually addressed in other operators.